### PR TITLE
[SPARK-45494][CORE][PYTHON] Introduce util functions to read/write a byte array for PythonWorkerUtils

### PR DESCRIPTION
**Pull request by ueshin**
### What changes were proposed in this pull request?

Introduce util functions to read/write a byte array for `PythonWorkerUtils`.

The following util functions will be added:

- `writeBytes`
- `readBytes`

### Why are the changes needed?

To read/write a byte array also happens often.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

The existing tests.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by ueshin** [comment]
Also cc @dtenedor @WweiL

File: core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala [hunk 1]
```diff
@@ -845,9 +843,7 @@ private[spark] class PythonRunner(
         try {
           stream.readInt() match {
             case length if length > 0 =>
-              val obj = new Array[Byte](length)
-              stream.readFully(obj)
-              obj
+              PythonWorkerUtils.readBytes(length, stream)
             case 0 => Array.emptyByteArray
```
Comment (allisonwang-db) [inline core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala:847]: Can we use `PythonWorkerUtils.readBytes` if we have `length=0` or we need to handle that separately?

**Review by zhengruifeng — APPROVED** [review]

File: core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala (hunk 1 above)
Comment (ueshin) [reply core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala:847]: That's a good point. I think we can include this in `readBytes` to always avoid creating a new empty array.

**Review by LuciferYang — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Pull request closed without merging**